From 629ff669a8813d8977b7fa2b60ee0cd4dd769f81 Mon Sep 17 00:00:00 2001
From: "Yue, VincentX" <vincentx.yue@intel.com>
Date: Wed, 20 Dec 2017 11:43:14 +0800
Subject: [PATCH 08/21] Fix system isn't responding when don't suport Bandwidth
 Control

Boot up time out due to don't support bandwith control,
it does not release the lock.

Change-Id: Idf5db6a3eb21635efe6a5bc6c44599efcc5150f9
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-53750
Signed-off-by: Yue, VincentX <vincentx.yue@intel.com>
---
 .../core/java/com/android/server/net/NetworkPolicyManagerService.java  | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
index 5159c70..726a2bf 100644
--- a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
+++ b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
@@ -794,9 +794,10 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
                     new NetworkRequest.Builder().build(), mNetworkCallback);
 
             mUsageStats.addAppIdleStateChangeListener(new AppIdleStateChangeListener());
+        } finally {
             // tell systemReady() that the service has been initialized
             initCompleteSignal.countDown();
-        } finally {
+
             // Restore the default priority after init is done
             Process.setThreadPriority(oldPriority);
             Trace.traceEnd(Trace.TRACE_TAG_NETWORK);
-- 
1.9.1

